src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -512,6 +509,42 @@ private CompletableFuture<Void> setPoNumberIfMissing(CompositePurchaseOrder comp
     return completedFuture(null);
   }
 
+  public void updatePrefixAndSuffix(CompositePurchaseOrder poFromStorage, CompositePurchaseOrder updatedPo) {
+    String poNumber = "";
+    if(HelperUtils.isPrefixChanged(poFromStorage, updatedPo)) {
+      if(Objects.nonNull(updatedPo.getPoNumberPrefix()) && !updatedPo.getPoNumberPrefix().isEmpty()) {
+        if(Objects.nonNull(poFromStorage.getPoNumberPrefix()) && updatedPo.getPoNumber().contains(poFromStorage.getPoNumberPrefix())) {
+          poNumber = updatedPo.getPoNumber().replaceAll(poFromStorage.getPoNumberPrefix(), "");
+          updatedPo.setPoNumber(updatedPo.getPoNumberPrefix() + poNumber);
+        }
+        else {
+          HelperUtils.setPoNumberPrefix(updatedPo);
+        }
+      }
+      else {
+        poNumber = updatedPo.getPoNumber().replaceAll(poFromStorage.getPoNumberPrefix(), "");
+        updatedPo.setPoNumber(poNumber);
+      }
+    }
+
+    if(HelperUtils.isSuffixChanged(poFromStorage, updatedPo)) {
+      if(Objects.nonNull(updatedPo.getPoNumberSuffix()) && !updatedPo.getPoNumberSuffix().isEmpty()) {
```

Comment:
Also the same approach to need to apply here

Reply:
fixed